### PR TITLE
Portfolio: preserve data in the browser

### PR DESCRIPTION
**Pull request by Tessyrich**
We added the necessary i.nteractions on the form as required in the project

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -237,3 +237,45 @@ formVal.addEventListener('submit', (e) => {
...
+const formInfo2 = JSON.parse(localStorage.getItem('formInfo'));
+if (formInfo2) {
+  Firstnameinput.value = formInfo2.Firstname;
+  Fullnameinput.value = formInfo2.Fullname;
+  Lastnameinput.value = formInfo2.Lastname;
+  Textareainput.value = formInfo2.Textarea;
+  Emailinput.value = formInfo2.Email;
+}
+
+formVal.addEventListener('submit', (event) => {
```
Comment (topeogunleye) [inline script.js:258]: ## Using a submit event handle will not let the data be saved in localStorage as the user is typing his inputs. An input event listener is better suited for this purpose. As required, Kindly ensure the following:

- [ ] After filling the form, and not submitting it, the entered values should be saved in the browser.
- [ ] The contact form information is persistent to page reloads.
- [ ] When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data. 

## EXAMPLE

```
form.addEventListener('input', () => {
  const data = {
    fullName: fullName.value,
    email: email.value,
    message: message.value,
  };

  localStorage.setItem('dataform', JSON.stringify(data));
});
```

File: script.js [hunk 2]
```diff
@@ -237,3 +237,45 @@ formVal.addEventListener('submit', (e) => {
...
+    Fullname,
+    Lastname,
+    Firstname,
+    Email,
+    Textarea,
+  };
+
+  // localStorage.setItem('formInfo', JSON.stringify(formInfo));
+  localStorage.setItem('formInfo', formInfo);
+});
```
Comment (topeogunleye) [inline script.js:281]: - [ ] Kindly ensure that when a user loads the page, the code checks if there's any saved data in the local storage and if so the input fields are pre-filled with the data.

## EXAMPLE
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```

File: index.html [hunk 3]
```diff
@@ -279,7 +279,7 @@ <h6 class="description">Hello I’m a software developer! I can help you build a
     </section>
   </div>
   <footer class="form-cont" id="form-cont">
-  <form action="https://formspree.io/f/mrgdkyla" method="post"  class="form">
+  <form class="form">
```
Comment (topeogunleye) [inline index.html:282]: - [ ] [OPTIONAL] I kindly suggest that you follow the Figma design template by ensuring that the text on the left side of your form is aligned like it is in the Figma Design.



|  |  |
| ---      | ---       |
| Your Design | image         |
| Figma Design    | image      |

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Tessyrich, and Team

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good description :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

## Hey @Tessyrich, @topeogunleye :wave:

Great work on implementing the localStorage. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

:heavy_check_mark: Following Requirements
✔️ Implemented the LocalStorage
:heavy_check_mark: Following Github Flow
:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Descriptive PR Summary

# Optional
- [ ] It is a pair-programming project so it should contain both partner commits. Kindly keep this in mind next time when you do a pair-programming project 🤔 


_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**